### ISSUE TITLE
labels for the the multi class classifier

### ISSUE THREAD

**Issue by naarkhoo** [Microsoft/LightGBM#1349]
Seems the response variable should be encoded as numeric. It is not intuitive in the the multi-class classifier, how the response variables are encoded, and [seems there is no way to access the model parameter](https://stackoverflow.com/questions/49674112/accessing-lightgbm-model-parameters)
  
Here is an example for the multi-class classification 

```
require(lightgbm)

# We load the default iris dataset shipped with R
data(iris)

# We must convert factors to numeric
# They must be starting from number 0 to use multiclass
# For instance: 0, 1, 2, 3, 4, 5...
iris$Species <- as.numeric(as.factor(iris$Species)) - 1

# We cut the data set into 80% train and 20% validation
# The 10 last samples of each class are for validation

train <- as.matrix(iris[c(1:40, 51:90, 101:140), ])
test <- as.matrix(iris[c(41:50, 91:100, 141:150), ])
dtrain <- lgb.Dataset(data = train[, 1:4], label = train[, 5])
dtest <- lgb.Dataset.create.valid(dtrain, data = test[, 1:4], label = test[, 5])
valids <- list(test = dtest)

# Method 1 of training
params <- list(objective = "multiclass", metric = "multi_error", num_class = 3)
model <- lgb.train(params,
                   dtrain,
                   100,
                   valids,
                   min_data = 1,
                   learning_rate = 1,
                   early_stopping_rounds = 10)

# We can predict on test data, outputs a 90-length vector
# Order: obs1 class1, obs1 class2, obs1 class3, obs2 class1, obs2 class2, obs2 class3...
my_preds <- predict(model, test[, 1:4])

# Method 2 of training, identical
model <- lgb.train(list(),
                   dtrain,
                   100,
                   valids,
                   min_data = 1,
                   learning_rate = 1,
                   early_stopping_rounds = 10,
                   objective = "multiclass",
                   metric = "multi_error",
                   num_class = 3)

# We can predict on test data, identical
my_preds <- predict(model, test[, 1:4])

# A (30x3) matrix with the predictions, use parameter reshape
# class1 class2 class3
#   obs1   obs1   obs1
#   obs2   obs2   obs2
#   ....   ....   ....
my_preds <- predict(model, test[, 1:4], reshape = TRUE)
```

So the size of `_my_preds_` matrix is (number_of_test_observation times number_of_classes) and each value correspond to the likelihood of the class (from the link function).

What is important to me, is basically, what are the column names ? e.g. what the first column (what is the first class label ? (is it the same order as `unique(test[, 5])` ?!

**Comment by Laurae2** [comment]
The number of the column corresponds to the number of the label minus 1.

* Column 1 = Label 0
* Column 2 = Label 1
* Column 3 = Label 2

**Comment by naarkhoo** [comment]
So it means, t is the same as factors from `as as.factor(iris$Species)` - the code above is just an example that labels are numbers.

**Comment by Laurae2** [comment]
It is mandatory to have numbers for the labels in LightGBM, starting from 0 to (n-1)th class if doing multiclass classification.

**Comment by naarkhoo** [comment]
Totally - I got that - it is just going back to the original data, and interpreting the result (so I needed to make sure, I am mapping-back correctly).

**Comment by laziszaire** [comment]
is it the same order as unique(test[, 5])?

**Comment by naarkhoo** [comment]
Yes
